<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1389</width>
    <height>1014</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>true</bool>
  </property>
  <property name="focusPolicy">
   <enum>Qt::StrongFocus</enum>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="mouseTracking">
    <bool>true</bool>
   </property>
   <property name="acceptDrops">
    <bool>true</bool>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout"/>
  </widget>
  <widget class="QToolBar" name="navigationToolBar">
   <property name="windowTitle">
    <string>navigationToolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="iconSize">
    <size>
     <width>25</width>
     <height>25</height>
    </size>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionFront"/>
   <addaction name="actionBack"/>
   <addaction name="actionTop"/>
   <addaction name="actionBottom"/>
   <addaction name="actionLeft"/>
   <addaction name="actionRight"/>
   <addaction name="actionIsometric"/>
   <addaction name="separator"/>
   <addaction name="actionZoom_In"/>
   <addaction name="actionZoom_Out"/>
   <addaction name="actionZoom_All"/>
   <addaction name="actionZoom_Window"/>
   <addaction name="separator"/>
   <addaction name="actionShade"/>
   <addaction name="actionMesh"/>
   <addaction name="actionNode"/>
   <addaction name="actionProfile"/>
   <addaction name="actionFaceNormal"/>
   <addaction name="actionNodeNormal"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="selectionToolBar">
   <property name="windowTitle">
    <string>selectionToolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="iconSize">
    <size>
     <width>25</width>
     <height>25</height>
    </size>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="separator"/>
   <addaction name="actionSelectNode"/>
   <addaction name="actionSelectEdge"/>
   <addaction name="actionSelectFace"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="8" column="0">
      <widget class="QTreeView" name="treeView">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>320</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>400</height>
        </size>
       </property>
       <property name="showDropIndicator" stdset="0">
        <bool>true</bool>
       </property>
       <property name="indentation">
        <number>5</number>
       </property>
       <attribute name="headerVisible">
        <bool>false</bool>
       </attribute>
      </widget>
     </item>
     <item row="7" column="0">
      <spacer name="verticalSpacer_2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="5" column="0">
      <widget class="QTabWidget" name="tabWidget">
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>500</height>
        </size>
       </property>
       <property name="currentIndex">
        <number>1</number>
       </property>
       <property name="iconSize">
        <size>
         <width>16</width>
         <height>16</height>
        </size>
       </property>
       <widget class="QWidget" name="tab">
        <attribute name="title">
         <string>Soft Finger With Ball</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QCheckBox" name="checkBox_combinedInput">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Combined Input</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>IterTime</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_iterTime">
             <property name="value">
              <number>1</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QCheckBox" name="checkBox_readChamberRegion">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Read Chamber Region</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="doubleSpinBox_A1">
           <property name="minimum">
            <double>1.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="value">
            <double>1.300000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_generateTETMesh">
           <property name="text">
            <string>Deformation &amp;&amp; Collision Response</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_runCollisionChecking">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Collision Response Testing</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_clearAll">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Clear All</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_planeDir">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <item>
            <property name="text">
             <string>X</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Y</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Z</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="horizontalSlider_slice_Multi_dir">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer_4">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>237</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_2">
        <attribute name="title">
         <string>Twisting Robot</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QPushButton" name="pushButton_inputFourChambers">
           <property name="text">
            <string>Input Twisting Robot</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_inputMem">
           <property name="text">
            <string>Input Obstacle</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_GenerateChamberTetMesh">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Generate Tet Mesh</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="Line" name="line">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Expansion Ratio</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="doubleSpinBox_Chamber1">
           <property name="minimum">
            <double>1.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="value">
            <double>4.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_ChamberDeformation">
           <property name="text">
            <string>Conduct Deformation</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_CollisionChecking">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Collision Checking &amp;&amp; Response</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="Line" name="line_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_CollisionResponse">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_trajectoryGeneration">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>142</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_3">
        <attribute name="title">
         <string>Obstacle Movement</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Move Obstacle</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="doubleSpinBox_rotationAngleX">
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="value">
            <double>0.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="doubleSpinBox_rotationAngleY">
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="value">
            <double>0.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="doubleSpinBox_rotationAngleZ">
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="value">
            <double>0.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_applyrotation">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>Apply Rotation</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="doubleSpinBox_ballMoveX">
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="value">
            <double>80.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="doubleSpinBox_ballMoveY">
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="value">
            <double>-40.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="doubleSpinBox_ballMoveZ">
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="value">
            <double>0.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_MoveBall">
           <property name="text">
            <string>Apply Translation</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer_3">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>319</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1389</width>
     <height>21</height>
    </rect>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionFront"/>
    <addaction name="actionBack"/>
    <addaction name="actionTop"/>
    <addaction name="actionBottom"/>
    <addaction name="actionLeft"/>
    <addaction name="actionRight"/>
    <addaction name="actionIsometric"/>
    <addaction name="separator"/>
    <addaction name="actionZoom_In"/>
    <addaction name="actionZoom_Out"/>
    <addaction name="actionZoom_All"/>
    <addaction name="actionZoom_Window"/>
    <addaction name="separator"/>
    <addaction name="actionShade"/>
    <addaction name="actionMesh"/>
    <addaction name="actionNode"/>
    <addaction name="actionProfile"/>
    <addaction name="separator"/>
    <addaction name="actionShifttoOrigin"/>
   </widget>
   <widget class="QMenu" name="menuSelect">
    <property name="title">
     <string>Select</string>
    </property>
    <addaction name="actionSelectNode"/>
    <addaction name="actionSelectEdge"/>
    <addaction name="actionSelectFace"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
   <addaction name="menuSelect"/>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Open Folder.png</normaloff>:/resource/Open Folder.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionFront">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Front View.png</normaloff>:/resource/Front View.png</iconset>
   </property>
   <property name="text">
    <string>Front</string>
   </property>
  </action>
  <action name="actionBack">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Back View.png</normaloff>:/resource/Back View.png</iconset>
   </property>
   <property name="text">
    <string>Back</string>
   </property>
  </action>
  <action name="actionTop">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Top View.png</normaloff>:/resource/Top View.png</iconset>
   </property>
   <property name="text">
    <string>Top</string>
   </property>
  </action>
  <action name="actionBottom">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Bottom View.png</normaloff>:/resource/Bottom View.png</iconset>
   </property>
   <property name="text">
    <string>Bottom</string>
   </property>
  </action>
  <action name="actionLeft">
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Left View.png</normaloff>:/resource/Left View.png</iconset>
   </property>
   <property name="text">
    <string>Left</string>
   </property>
  </action>
  <action name="actionRight">
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Right View.png</normaloff>:/resource/Right View.png</iconset>
   </property>
   <property name="text">
    <string>Right</string>
   </property>
  </action>
  <action name="actionIsometric">
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Isometric View.png</normaloff>:/resource/Isometric View.png</iconset>
   </property>
   <property name="text">
    <string>Isometric</string>
   </property>
  </action>
  <action name="actionZoom_In">
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Zoom In.png</normaloff>:/resource/Zoom In.png</iconset>
   </property>
   <property name="text">
    <string>Zoom In</string>
   </property>
  </action>
  <action name="actionZoom_Out">
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Zoom Out.png</normaloff>:/resource/Zoom Out.png</iconset>
   </property>
   <property name="text">
    <string>Zoom Out</string>
   </property>
  </action>
  <action name="actionZoom_All">
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Zoom All.png</normaloff>:/resource/Zoom All.png</iconset>
   </property>
   <property name="text">
    <string>Zoom All</string>
   </property>
  </action>
  <action name="actionZoom_Window">
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Zoom Window.png</normaloff>:/resource/Zoom Window.png</iconset>
   </property>
   <property name="text">
    <string>Zoom Window</string>
   </property>
  </action>
  <action name="actionShade">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Shade.png</normaloff>:/resource/Shade.png</iconset>
   </property>
   <property name="text">
    <string>Shade</string>
   </property>
  </action>
  <action name="actionMesh">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Mesh.png</normaloff>:/resource/Mesh.png</iconset>
   </property>
   <property name="text">
    <string>Mesh</string>
   </property>
  </action>
  <action name="actionNode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Node.png</normaloff>:/resource/Node.png</iconset>
   </property>
   <property name="text">
    <string>Node</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Save as.png</normaloff>:/resource/Save as.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionSelectNode">
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/selectNode.png</normaloff>:/resource/selectNode.png</iconset>
   </property>
   <property name="text">
    <string>Node</string>
   </property>
  </action>
  <action name="actionSelectFace">
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/selectFace.png</normaloff>:/resource/selectFace.png</iconset>
   </property>
   <property name="text">
    <string>Face</string>
   </property>
  </action>
  <action name="actionShifttoOrigin">
   <property name="text">
    <string>Shift to Origin</string>
   </property>
  </action>
  <action name="actionProfile">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/Profile.png</normaloff>:/resource/Profile.png</iconset>
   </property>
   <property name="text">
    <string>Profile</string>
   </property>
  </action>
  <action name="actionFaceNormal">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/FaceNormal.png</normaloff>:/resource/FaceNormal.png</iconset>
   </property>
   <property name="text">
    <string>FaceNormal</string>
   </property>
  </action>
  <action name="actionNodeNormal">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/NodeNormal.png</normaloff>:/resource/NodeNormal.png</iconset>
   </property>
   <property name="text">
    <string>NodeNormal</string>
   </property>
  </action>
  <action name="actionSelectEdge">
   <property name="icon">
    <iconset resource="ShapeLab.qrc">
     <normaloff>:/resource/selectEdge.png</normaloff>:/resource/selectEdge.png</iconset>
   </property>
   <property name="text">
    <string>Edge</string>
   </property>
  </action>
  <action name="actionGenerate">
   <property name="text">
    <string>Generate</string>
   </property>
  </action>
  <action name="actionTest_1">
   <property name="text">
    <string>Test_1</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="ShapeLab.qrc"/>
 </resources>
 <connections/>
</ui>
